### PR TITLE
add node-redlock

### PR DESCRIPTION
**Pull request by danbrakeley**
- [x] @lpetre

### CODE REVIEW

File: src/redlock/Redlock.hx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+extern class Lock {
+  public function unlock(cb:Void->Void) : Void;
+  public function extend(ttl:Int, cb:LockCallback) : Void;
+}
+
+@:jsRequire('redlock')
+extern class Redlock {
+  public function new(redisClients:Array<Dynamic>, options:Options);
```
Comment (lpetre) [inline src/redlock/Redlock.hx]: Array<redis.RedisClient> ?

File: src/redlock/Redlock.hx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  public function lock(resource:String, ttl:Int, cb:LockCallback) : Void;
+  public function unlock(lock:Lock, cb:Void->Void) : Void;
```
Comment (lpetre) [inline src/redlock/Redlock.hx]: Interesting unlock returns no information.

**Comment by joemukai** [comment]
Please update README.md

File: src/redlock/Redlock.hx (hunk 1 above)
Comment (danbrakeley) [inline src/redlock/Redlock.hx]: It felt gross to import redis.Redis inside this file, but maybe we're OK with that?

File: src/redlock/Redlock.hx (hunk 2 above)
Comment (danbrakeley) [inline src/redlock/Redlock.hx]: Yeah, that weirded me out, too. Shouldn't that somehow report an error if the random number in the lock doesn't match up?

**Comment by danbrakeley** [comment]
done.

File: src/redlock/Redlock.hx (hunk 1 above)
Comment (danbrakeley) [inline src/redlock/Redlock.hx]: Well, I did it. I don't have a strong opinion on it, really.

**Pull request merged**